### ISSUE TITLE
Maybe let's join forces to update vacuum gem?

### ISSUE THREAD

**Issue by skatkov** [dkam/paapi#2]
I've been looking if anyone has attempted to write a ruby wrapper for new PA-API and found your library. Great work, thanks for building this in an open.

I reviewed your code, but I don't think this library will fit my use-case:
- It's really hard to come up with response object parser that fits everyone's use case.
- I didn't felt like having yeat-another-http-adapter to increase my projects dependency (so i opted for HTTPi , so anyone can roll with their own adapter)
- It seems, that providing DEFAULT_RESOURCE or [modifying resources](https://github.com/dkam/paapi/blob/master/lib/paapi/client.rb#L67) within a library is again, quite a opinionated move that will not fit every usecase

In my personal opinion, Vacuum was almost-perfect implementation for PA-API version 4 and had quite a following already. I attempted to re-implement Vacuum in this pull request:
https://github.com/hakanensari/vacuum/pull/78

I'd love to get your input. Maybe you'll be in helping re-implement gem that already has heavily usage? :-)

Thanks again!

**Comment by dkam** [comment]
Hey thanks for the comments!  I've not seen the vacuum gem - I've been using the amazon-ecs gem so thanks for the tip. 

To address your points: 
 * The parsing of the response into items has been stumping me - I don't really want to have to enumerate ever possible resource type.  I see Vacuum just returns the hash, but provides a way to add your own parser. I've experimented with returning OpenStruct to allow dot navigation, but it just doesn't smell nice. 
 * The HTTP request is made in a single file, changing it is very easy and I'm happy to take feedback. The HTTPrb gem seems good, but I've also been happy with HTTPClient. I'll take a look at HTTPi. 
 * The get_browse_nodes & get_variations methods are place holder until I finish the methods I use personally - get_items and search_items.  

This gem has, as you've seen, a few bits of dead end / exploration code in it that I need to clean up. 

Thanks again for the review and for pointing me to Vacuum - I'll take a close look, but my quick glance over it looks good.

**Comment by skatkov** [comment]
@hakanensary was kind enough to merge my unfinished implementation in v5 branch. 

https://github.com/hakanensari/vacuum/tree/v5

Since original implementation was quite minimal and not opinionated -- I've rebuilt it with same principals. So while using vacuum gem in my own project, I managed to build up things on top of it -- my own error handler and parser. 

Hopefully, it should fit your use case as well.

Here are couple of updates:
- Unfortunately, I don't need `search_items` in my project -- so it's the only method that is not covered
- We ended up dropping HTTPi to have less dependencies and just rely on ruby's Net::HTTP
